src/NuGet.Clients/NuGet.VisualStudio/Extensibility/IVsPathContext.cs

```diff
@@ -24,5 +26,33 @@ public interface IVsPathContext
         /// <see cref="string"/>.
         /// </summary>
         IEnumerable FallbackPackageFolders { get; }
+
+        /// <summary>
+        /// Fetch a package directory containing the provided asset path.
+        /// </summary>
+        /// <param name="referenceFilePath">Absolute path to package asset file.</param>
+        /// <param name="packageDirectoryPath">Full path to a package directory. 
+        /// <code>null</code> if returned falue is <code>false</code>.</param>
+        /// <returns>
+        /// <code>true</code> when a package containing the given file was found, <code>false</code> - otherwise.
+        /// </returns>
+        /// <example>
+        /// Suppose the project is a packages.config project and the following asset paths are provided:
+        /// 
+        /// - C:\src\MyProject\packages\NuGet.Versioning.3.5.0-rc1-final\lib\net45\NuGet.Versioning.dll
+        /// - C:\path\to\non\package\assembly\Newtonsoft.Json.dll
+        /// - C:\src\MyOtherProject\packages\NuGet.Core.2.12.0\lib\net40\NuGet.Core.dll
+        /// - C:\src\MyProject\packages\Autofac.3.5.2\lib\net40\Autofac.dll
+        /// - C:\src\MyProject\packages\Autofac.3.5.2\lib\net40\Autofac.Fake.dll
+        /// 
+        /// The result will be:
+        /// 
+        /// - C:\src\MyProject\packages\NuGet.Versioning.3.5.0-rc1-final
+        /// - null
+        /// - null
+        /// - C:\src\MyProject\packages\Autofac.3.5.2
+        /// - C:\src\MyProject\packages\Autofac.3.5.2
+        /// </example>
+        bool TryResolveReference(string referenceFilePath, out string packageDirectoryPath);
```

Comment:
Maybe the referenceFilePath parameter should be renamed to something more generic?  From the description it sounds like the caller could provide the path to any file (maybe even a non-existent file?) and they're not limited to reference assemblies / DLLs?

That also makes me think that TryResolveReference might not be the right name.